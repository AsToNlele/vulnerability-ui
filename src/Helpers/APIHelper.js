let BASE_ROUTE = '/api/v1';

export function createApiCall(endpoint, method, parameters = undefined, data = undefined) {
    if (parameters && method === 'get') {
        let params = Object.keys(parameters)
        .map(key => [
            encodeURIComponent(key)
            .concat('=')
            .concat(encodeURIComponent(parameters[key]))
        ])
        .join('&');
        endpoint = endpoint.concat('?').concat(params);
    }

    return fetch('https://localhost:8300' + BASE_ROUTE + endpoint, {
        method,
        body: JSON.stringify(data)
    }).then(function(res) {
        if (!res.ok) {
            throw new Error(res.statusText);
        }

        return res.json();
    });
}

export function getCveDetails(synopsis) {
    let endpoint = '/cves/' + synopsis + '/details/';
    let result = createApiCall(endpoint, 'get');
    return result;
}

export function getAffectedSystemsByCVE(synopsis) {
    let endpoint = '/cves/' + synopsis + '/affectedsystems/';
    let result = createApiCall(endpoint, 'get');
    return result;
}

export function getChanges() {
    let endpoint = '/vulnerabilities/changes/';
    let result = createApiCall(endpoint, 'get');
    return result;
}

export function getCurrentImpacts() {
    let endpoint = '/vulnerabilities/impacts/';
    let result = createApiCall(endpoint, 'get');
    return result;
}

export function getCveList(apiProps) {
    let defaultProps = {
        sortBy: '',
        filter: '',
        cvss_score: '',
        publicDate: ''
    };
    let parameters = { ...defaultProps, ...apiProps };
    let endpoint = '/vulnerabilities/cves/';
    let result = createApiCall(endpoint, 'get', parameters);
    return result;
}
