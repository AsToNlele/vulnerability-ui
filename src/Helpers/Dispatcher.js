import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
import ReducerRegistry from '../Utilities/ReducerRegistry';
export function dispatchAction(actionCreator) {
    const store = ReducerRegistry.getStore();
    return store.dispatch(actionCreator).catch(res => {
        store.dispatch(
            addNotification({
                variant: 'danger',
                title: 'Error ' + res.status,
                description: res.statusText
            })
        );
    });
}
