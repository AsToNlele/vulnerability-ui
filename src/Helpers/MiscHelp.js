export function dispatchAction(actionCreator, dispatch) {
    return dispatch(actionCreator).catch(() => {});
}
