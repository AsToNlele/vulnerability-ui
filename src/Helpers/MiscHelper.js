import { ExclamationTriangleIcon } from '@patternfly/react-icons';
import propTypes from 'prop-types';
import React from 'react';

/* global require */

export function processDate(dateString) {
    let date = new Date(dateString);
    let dateFormat = require('dateformat');
    return dateFormat(date, 'yyyy-mm-dd');
}

export function processImpactForTable(impact) {
    let icon;
    switch (impact) {
        case 'High':
        case 'Important':
            icon = <ExclamationTriangleIcon size="md" color="#ec7a08" />;
            break;
        case 'Medium':
        case 'Moderate':
            icon = <ExclamationTriangleIcon size="md" color="var(--pf-global--warning-color--100)" />;
            break;
        case 'Critical':
            icon = <ExclamationTriangleIcon size="md" color="var(--pf-global--danger-color--100)" />;
            break;
        case 'Low':
            icon = <ExclamationTriangleIcon size="md" color="var(--pf-global--BackgroundColor--disabled)" />;
            break;
    }

    return <span>{icon}</span>;
}

export function processCvssScore(score) {
    return score || 'N/A';
}

export function truncate(str, max) {
    return str.length > max ? str.substr(0, max - 1) + '...' : str;
}

export const dataShape = propTypes.shape({
    data: propTypes.oneOfType([propTypes.object, propTypes.array]),
    meta: propTypes.object,
    isLoading: propTypes.bool
});
