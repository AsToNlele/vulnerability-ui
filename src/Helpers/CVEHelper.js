import { LinkIcon } from '@patternfly/react-icons';
import React from 'react';
import { createApiCall } from './APIHelper';
import * as MiscHelper from './MiscHelper';

export function getCveDetails(synopsis) {
    let endpoint = '/cves/' + synopsis + '/details/';
    let result = createApiCall(endpoint, 'get');
    return result;
}

export function getAffectedSystemsByCVE(synopsis) {
    let endpoint = '/cves/' + synopsis + '/affectedsystems/';
    let result = createApiCall(endpoint, 'get');
    return result;
}

export function createExposedSystemsTable(affectedSystems) {
    // TODO: Use correct API format instead this one
    let isLoading = affectedSystems.isLoading;
    if (!isLoading) {
        let payload = affectedSystems.payload.asMutable();
        let data = payload;
        let meta = {};
        return { data, meta, isLoading };
    }

    return { data: [], meta: {}, isLoading };
}

export function createCveDetailsPage(cves) {
    let dataSet = cves.payload;
    let isLoading = cves.isLoading;
    if (!isLoading) {
        let singleCve = cves.payload;
        dataSet = {
            impact: MiscHelper.processImpactForTable(singleCve.impact),
            public_date: MiscHelper.processDate(singleCve.public_date),
            description: singleCve.description,
            synopsis: singleCve.synopsis,
            errata_list: singleCve.errata_list.map(item => <li key={item}>{createErrataLink(item)}</li>),
            package_list: singleCve.package_list.map(item => <li key={item}>{item}</li>),
            url_list: linkToDatabase(singleCve.synopsis),
            cvss3_score: singleCve.cvss3_score
        };
    }

    dataSet = { content: dataSet, isLoading };
    return dataSet;
}

function createErrataLink(item) {
    return <a href={'https://access.redhat.com/errata/' + item}>{item}</a>;
}

function linkToDatabase(cve) {
    let link = 'https://access.redhat.com/security/cve/' + cve;
    return (
        <a href={link}>
            <LinkIcon title="link" /> link
        </a>
    );
}
