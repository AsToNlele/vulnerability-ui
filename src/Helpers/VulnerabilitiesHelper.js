import * as MiscHelper from './MiscHelper';

export function createCveListByAccount(cveList) {
    let isLoading = cveList.isLoading;
    if (!isLoading) {
        let payload = cveList.payload.asMutable();
        let data = payload.data;
        let meta = payload.meta;
        data = data.map(row => ({
            id: row.id,
            cells: [
                MiscHelper.processImpactForTable(row.attributes.impact),
                row.attributes.synopsis,
                MiscHelper.truncate(row.attributes.description, 200),
                row.attributes.cvss_score,
                row.attributes.systems_affected,
                'No',
                MiscHelper.processDate(row.attributes.public_date)
            ]
        }));
        return { data, meta, isLoading };
    }

    return { data: [], meta: {}, isLoading };
}

export function createCveListBySystem(cveList) {
    let isLoading = cveList.isLoading;
    if (!isLoading) {
        let payload = cveList.payload.asMutable();
        let data = payload.data;
        let meta = payload.meta;
        data = data.map(row => ({
            id: row.id,
            cells: [
                MiscHelper.processImpactForTable(row.attributes.impact),
                row.attributes.synopsis,
                MiscHelper.truncate(row.attributes.description, 200),
                row.attributes.cvss_score,
                MiscHelper.processDate(row.attributes.public_date)
            ]
        }));
        return { data, meta, isLoading };
    }

    return { data: [], meta: {}, isLoading };
}

export function createCurrentVulnerabilitiesCard(impacts) {
    let dataSet = impacts.payload;
    let isLoading = impacts.isLoading;
    if (!isLoading) {
        dataSet = impacts.payload;
    }

    dataSet = { content: dataSet, isLoading };
    return dataSet;
}
