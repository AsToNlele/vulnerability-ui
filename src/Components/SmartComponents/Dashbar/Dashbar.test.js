import Dashbar from "./Dashbar";
import { Card, Grid } from "@patternfly/react-core";
import toJson from "enzyme-to-json";
import { mountWithIntl } from "../../../Helpers/MiscHelper";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import { DashbarItem } from './Dashbar'

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
}));

const mockStore = configureStore([(store) => (next) => (action) => { }]);
let store = mockStore({});

jest.mock("../../../Helpers/MiscHelper", () => ({
    ...jest.requireActual("../../../Helpers/MiscHelper"),
    useUrlParams: () => [{ dashbar: "true" }, jest.fn()],
}));


jest.mock("../../../Helpers/APIHelper", () => ({
    ...jest.requireActual("../../../Helpers/APIHelper"),
    getAnnouncement: () => new Promise((resolve) => resolve({ message: "Message test!" })),
    getDashbar: () => new Promise((resolve) => resolve({ exploitable_cves: 1, cves_with_rule: 2, critical_cves: 3, important_cves: 4 }))
}))

describe("Dashbar", () => {
    it("Should match the snapshot", async () => {

        const wrapper = await mountWithIntl(
            <Provider store={store}>
                <Router>
                    <Dashbar />
                </Router>
            </Provider>
        );
        wrapper.update()
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it("Should render Grid with props hasGutter = true", async () => {
        const wrapper = await mountWithIntl(
            <Provider store={store}>
                <Router>
                    <Dashbar />
                </Router>
            </Provider>
        );
        wrapper.update()
        expect(wrapper.find(Grid)).toHaveLength(1);
        expect(wrapper.find(Grid).prop("hasGutter")).toBeTruthy();
    });
    it("Should have 4 DashbarItems", async () => {
        const wrapper = await mountWithIntl(
            <Provider store={store}>
                <Router>
                    <Dashbar />
                </Router>
            </Provider>
        );
        wrapper.update()
        expect(wrapper.find(DashbarItem)).toHaveLength(4);
    });
});
