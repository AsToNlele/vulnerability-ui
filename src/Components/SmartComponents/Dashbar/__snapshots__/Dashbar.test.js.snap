// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashbar Should match the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Dashbar>
        <Connect(Main)
          style={
            Object {
              "paddingBottom": 0,
            }
          }
        >
          <Main
            style={
              Object {
                "paddingBottom": 0,
              }
            }
          >
            <section
              className="pf-l-page__main-section pf-c-page__main-section"
              page-type=""
              style={
                Object {
                  "paddingBottom": 0,
                }
              }
            >
              <Stack
                hasGutter={true}
              >
                <div
                  className="pf-l-stack pf-m-gutter"
                >
                  <StackItem>
                    <div
                      className="pf-l-stack__item"
                    >
                      <Grid
                        hasGutter={true}
                      >
                        <div
                          className="pf-l-grid pf-m-gutter"
                        >
                          <GridItem
                            md={3}
                            span={12}
                          >
                            <div
                              className="pf-l-grid__item pf-m-12-col pf-m-3-col-on-md"
                            >
                              <DashbarItem
                                count={1}
                                onLinkClick={[Function]}
                                title={
                                  <Memo(MemoizedFormattedMessage)
                                    defaultMessage="CVEs with known exploits"
                                    description="Title of Known exploit card"
                                    id="dashbarKnownExploitsTitle"
                                  />
                                }
                              >
                                <Card
                                  className="card-box"
                                  isFullHeight={true}
                                >
                                  <article
                                    className="pf-c-card pf-m-full-height card-box"
                                    data-ouia-component-id="OUIA-Generated-Card-1"
                                    data-ouia-component-type="PF4/Card"
                                    data-ouia-safe={true}
                                    id=""
                                  >
                                    <CardBody>
                                      <div
                                        className="pf-c-card__body"
                                      >
                                        <Text
                                          className="pf-u-font-size-lg pf-u-font-weight-bold"
                                        >
                                          <p
                                            className="pf-u-font-size-lg pf-u-font-weight-bold"
                                            data-ouia-component-id="OUIA-Generated-Text-1"
                                            data-ouia-component-type="PF4/Text"
                                            data-ouia-safe={true}
                                            data-pf-content={true}
                                          >
                                            <FormattedMessage
                                              defaultMessage="CVEs with known exploits"
                                              description="Title of Known exploit card"
                                              id="dashbarKnownExploitsTitle"
                                            >
                                              CVEs with known exploits
                                            </FormattedMessage>
                                          </p>
                                        </Text>
                                        <Text
                                          className="pf-u-font-size-lg pf-u-font-weight-bold pf-u-mt-xs"
                                        >
                                          <p
                                            className="pf-u-font-size-lg pf-u-font-weight-bold pf-u-mt-xs"
                                            data-ouia-component-id="OUIA-Generated-Text-2"
                                            data-ouia-component-type="PF4/Text"
                                            data-ouia-safe={true}
                                            data-pf-content={true}
                                          >
                                            <a
                                              onClick={[Function]}
                                            >
                                              1
                                            </a>
                                          </p>
                                        </Text>
                                      </div>
                                    </CardBody>
                                  </article>
                                </Card>
                              </DashbarItem>
                            </div>
                          </GridItem>
                          <GridItem
                            md={3}
                            span={12}
                          >
                            <div
                              className="pf-l-grid__item pf-m-12-col pf-m-3-col-on-md"
                            >
                              <DashbarItem
                                count={2}
                                onLinkClick={[Function]}
                                title={
                                  <Memo(MemoizedFormattedMessage)
                                    defaultMessage="CVEs with security rules"
                                    description="Title of Security rule card"
                                    id="dashbarSecurityRulesTitle"
                                  />
                                }
                              >
                                <Card
                                  className="card-box"
                                  isFullHeight={true}
                                >
                                  <article
                                    className="pf-c-card pf-m-full-height card-box"
                                    data-ouia-component-id="OUIA-Generated-Card-2"
                                    data-ouia-component-type="PF4/Card"
                                    data-ouia-safe={true}
                                    id=""
                                  >
                                    <CardBody>
                                      <div
                                        className="pf-c-card__body"
                                      >
                                        <Text
                                          className="pf-u-font-size-lg pf-u-font-weight-bold"
                                        >
                                          <p
                                            className="pf-u-font-size-lg pf-u-font-weight-bold"
                                            data-ouia-component-id="OUIA-Generated-Text-3"
                                            data-ouia-component-type="PF4/Text"
                                            data-ouia-safe={true}
                                            data-pf-content={true}
                                          >
                                            <FormattedMessage
                                              defaultMessage="CVEs with security rules"
                                              description="Title of Security rule card"
                                              id="dashbarSecurityRulesTitle"
                                            >
                                              CVEs with security rules
                                            </FormattedMessage>
                                          </p>
                                        </Text>
                                        <Text
                                          className="pf-u-font-size-lg pf-u-font-weight-bold pf-u-mt-xs"
                                        >
                                          <p
                                            className="pf-u-font-size-lg pf-u-font-weight-bold pf-u-mt-xs"
                                            data-ouia-component-id="OUIA-Generated-Text-4"
                                            data-ouia-component-type="PF4/Text"
                                            data-ouia-safe={true}
                                            data-pf-content={true}
                                          >
                                            <a
                                              onClick={[Function]}
                                            >
                                              2
                                            </a>
                                          </p>
                                        </Text>
                                      </div>
                                    </CardBody>
                                  </article>
                                </Card>
                              </DashbarItem>
                            </div>
                          </GridItem>
                          <GridItem
                            md={3}
                            span={12}
                          >
                            <div
                              className="pf-l-grid__item pf-m-12-col pf-m-3-col-on-md"
                            >
                              <DashbarItem
                                count={3}
                                hasIcon={true}
                                impact="Critical"
                                onLinkClick={[Function]}
                                title={
                                  <Memo(MemoizedFormattedMessage)
                                    defaultMessage="CVEs with critical severity"
                                    description="Title of Critical severity card"
                                    id="dashbarCriticalSeverityTitle"
                                  />
                                }
                              >
                                <Card
                                  className="card-box"
                                  isFullHeight={true}
                                >
                                  <article
                                    className="pf-c-card pf-m-full-height card-box"
                                    data-ouia-component-id="OUIA-Generated-Card-3"
                                    data-ouia-component-type="PF4/Card"
                                    data-ouia-safe={true}
                                    id=""
                                  >
                                    <CardBody>
                                      <div
                                        className="pf-c-card__body"
                                      >
                                        <Text
                                          className="pf-u-font-size-lg pf-u-font-weight-bold"
                                        >
                                          <p
                                            className="pf-u-font-size-lg pf-u-font-weight-bold"
                                            data-ouia-component-id="OUIA-Generated-Text-5"
                                            data-ouia-component-type="PF4/Text"
                                            data-ouia-safe={true}
                                            data-pf-content={true}
                                          >
                                            <FormattedMessage
                                              defaultMessage="CVEs with critical severity"
                                              description="Title of Critical severity card"
                                              id="dashbarCriticalSeverityTitle"
                                            >
                                              CVEs with critical severity
                                            </FormattedMessage>
                                          </p>
                                        </Text>
                                        <Text
                                          className="pf-u-font-size-lg pf-u-font-weight-bold pf-u-mt-xs"
                                        >
                                          <p
                                            className="pf-u-font-size-lg pf-u-font-weight-bold pf-u-mt-xs"
                                            data-ouia-component-id="OUIA-Generated-Text-6"
                                            data-ouia-component-type="PF4/Text"
                                            data-ouia-safe={true}
                                            data-pf-content={true}
                                          >
                                            <a
                                              onClick={[Function]}
                                            >
                                              <SecurityIcon
                                                className="pf-u-mr-sm"
                                                color="var(--pf-global--palette--red-200)"
                                                noVerticalAlign={false}
                                                size="sm"
                                              >
                                                <svg
                                                  aria-hidden={true}
                                                  aria-labelledby={null}
                                                  className="pf-u-mr-sm"
                                                  fill="var(--pf-global--palette--red-200)"
                                                  height="1em"
                                                  role="img"
                                                  style={
                                                    Object {
                                                      "verticalAlign": "-0.125em",
                                                    }
                                                  }
                                                  viewBox="0 0 896 1024"
                                                  width="1em"
                                                >
                                                  <path
                                                    d="M861.5,0 L34.5,0 C15.4,0 0,14.3 0,32 L0,452.1 C0,768 387.7,1024 448.5,1024 C509.3,1024 896,768 896,452.2 L896,32 C896,14.3 880.6,0 861.5,0 Z M490.7,768 L405.3,768 C393.5,767.8 384.2,757.5 384,744.7 L384,663.3 C384.2,650.5 393.6,640.3 405.3,640 L490.7,640 C502.5,640.2 511.8,650.5 512,663.3 L512,744.7 L512.1,744.7 C511.8,757.5 502.4,767.8 490.7,768 Z M543.9,162.7 L517.2,514.4 C515.8,530.9 502,544 485.3,544 L410.6,544 C394,544 380.1,531.2 378.7,514.7 L352.1,163 C350.5,144.3 365.3,128.3 384,128.3 L512,128 C530.7,128 545.4,144 543.9,162.7 Z"
                                                  />
                                                </svg>
                                              </SecurityIcon>
                                              3
                                            </a>
                                          </p>
                                        </Text>
                                      </div>
                                    </CardBody>
                                  </article>
                                </Card>
                              </DashbarItem>
                            </div>
                          </GridItem>
                          <GridItem
                            md={3}
                            span={12}
                          >
                            <div
                              className="pf-l-grid__item pf-m-12-col pf-m-3-col-on-md"
                            >
                              <DashbarItem
                                count={4}
                                hasIcon={true}
                                impact="Important"
                                onLinkClick={[Function]}
                                title={
                                  <Memo(MemoizedFormattedMessage)
                                    defaultMessage="CVEs with important severity"
                                    description="Title of Important Severity card"
                                    id="dashbarImportantSeverityTitle"
                                  />
                                }
                              >
                                <Card
                                  className="card-box"
                                  isFullHeight={true}
                                >
                                  <article
                                    className="pf-c-card pf-m-full-height card-box"
                                    data-ouia-component-id="OUIA-Generated-Card-4"
                                    data-ouia-component-type="PF4/Card"
                                    data-ouia-safe={true}
                                    id=""
                                  >
                                    <CardBody>
                                      <div
                                        className="pf-c-card__body"
                                      >
                                        <Text
                                          className="pf-u-font-size-lg pf-u-font-weight-bold"
                                        >
                                          <p
                                            className="pf-u-font-size-lg pf-u-font-weight-bold"
                                            data-ouia-component-id="OUIA-Generated-Text-7"
                                            data-ouia-component-type="PF4/Text"
                                            data-ouia-safe={true}
                                            data-pf-content={true}
                                          >
                                            <FormattedMessage
                                              defaultMessage="CVEs with important severity"
                                              description="Title of Important Severity card"
                                              id="dashbarImportantSeverityTitle"
                                            >
                                              CVEs with important severity
                                            </FormattedMessage>
                                          </p>
                                        </Text>
                                        <Text
                                          className="pf-u-font-size-lg pf-u-font-weight-bold pf-u-mt-xs"
                                        >
                                          <p
                                            className="pf-u-font-size-lg pf-u-font-weight-bold pf-u-mt-xs"
                                            data-ouia-component-id="OUIA-Generated-Text-8"
                                            data-ouia-component-type="PF4/Text"
                                            data-ouia-safe={true}
                                            data-pf-content={true}
                                          >
                                            <a
                                              onClick={[Function]}
                                            >
                                              <SecurityIcon
                                                className="pf-u-mr-sm"
                                                color="var(--pf-global--palette--orange-300)"
                                                noVerticalAlign={false}
                                                size="sm"
                                              >
                                                <svg
                                                  aria-hidden={true}
                                                  aria-labelledby={null}
                                                  className="pf-u-mr-sm"
                                                  fill="var(--pf-global--palette--orange-300)"
                                                  height="1em"
                                                  role="img"
                                                  style={
                                                    Object {
                                                      "verticalAlign": "-0.125em",
                                                    }
                                                  }
                                                  viewBox="0 0 896 1024"
                                                  width="1em"
                                                >
                                                  <path
                                                    d="M861.5,0 L34.5,0 C15.4,0 0,14.3 0,32 L0,452.1 C0,768 387.7,1024 448.5,1024 C509.3,1024 896,768 896,452.2 L896,32 C896,14.3 880.6,0 861.5,0 Z M490.7,768 L405.3,768 C393.5,767.8 384.2,757.5 384,744.7 L384,663.3 C384.2,650.5 393.6,640.3 405.3,640 L490.7,640 C502.5,640.2 511.8,650.5 512,663.3 L512,744.7 L512.1,744.7 C511.8,757.5 502.4,767.8 490.7,768 Z M543.9,162.7 L517.2,514.4 C515.8,530.9 502,544 485.3,544 L410.6,544 C394,544 380.1,531.2 378.7,514.7 L352.1,163 C350.5,144.3 365.3,128.3 384,128.3 L512,128 C530.7,128 545.4,144 543.9,162.7 Z"
                                                  />
                                                </svg>
                                              </SecurityIcon>
                                              4
                                            </a>
                                          </p>
                                        </Text>
                                      </div>
                                    </CardBody>
                                  </article>
                                </Card>
                              </DashbarItem>
                            </div>
                          </GridItem>
                        </div>
                      </Grid>
                    </div>
                  </StackItem>
                  <StackItem>
                    <div
                      className="pf-l-stack__item"
                    >
                      <Alert
                        isInline={true}
                        title="Announcement"
                        variant="warning"
                      >
                        <div
                          aria-label="Warning Alert"
                          className="pf-c-alert pf-m-inline pf-m-warning"
                          data-ouia-component-id="OUIA-Generated-Alert-warning-1"
                          data-ouia-component-type="PF4/Alert"
                          data-ouia-safe={true}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <AlertIcon
                            variant="warning"
                          >
                            <div
                              className="pf-c-alert__icon"
                            >
                              <ExclamationTriangleIcon
                                color="currentColor"
                                noVerticalAlign={false}
                                size="sm"
                              >
                                <svg
                                  aria-hidden={true}
                                  aria-labelledby={null}
                                  fill="currentColor"
                                  height="1em"
                                  role="img"
                                  style={
                                    Object {
                                      "verticalAlign": "-0.125em",
                                    }
                                  }
                                  viewBox="0 0 576 512"
                                  width="1em"
                                >
                                  <path
                                    d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                                  />
                                </svg>
                              </ExclamationTriangleIcon>
                            </div>
                          </AlertIcon>
                          <h4
                            className="pf-c-alert__title"
                          >
                            <span
                              className="pf-u-screen-reader"
                            >
                              Warning alert:
                            </span>
                            Announcement
                          </h4>
                          <div
                            className="pf-c-alert__description"
                          >
                            Message test!
                          </div>
                        </div>
                      </Alert>
                    </div>
                  </StackItem>
                </div>
              </Stack>
            </section>
          </Main>
        </Connect(Main)>
      </Dashbar>
    </Router>
  </BrowserRouter>
</Provider>
`;
